dev/camdenhells/REQUIREMENTS.md

```diff
@@ -55,34 +55,37 @@ For reviews and updates:
 ## 3.1 Android and OpenGL ES integration
 
 1. Provide (major and minor) version query for the current GL ES render context
-2. Allow multiple `GlSurfaceViews` per Activity ? [2.1]
-  * Is this actually worthwhile?
-  * Only one render thread/context/Renderer can exist at once, so SurfaceViews are mutually exclusive 
-    * E.g. `ViewPager` support (per #1619)
-    * Is timing of render context switch still based on window attachment?
-3. Wallpaper surface view multisampling fix (#1559)
-4. Provide query/configuration constants (bit flags) for all nVidia GL ES extensions (#1742) [2.1]
-  * Do any other vendors or extension categories need this?
+2. Decouple Android views from the EGL context to allow offline rendering/initialization
+3. ~Allow multiple `GlSurfaceViews` per Activity ? [2.1]~
+  ~* Is this actually worthwhile?~
+  ~* Only one render thread/context/Renderer can exist at once, so SurfaceViews are mutually exclusive~
+    ~* E.g. `ViewPager` support (per #1619)~
+    ~* Is timing of render context switch still based on window attachment?~
+4. Wallpaper surface view multisampling fix (#1559)
+5. Provide query/configuration constants (bit flags) for all Vendor GL ES extensions (#1742) [2.1]
+  * CH implementation will include this for all texture extensions.
   
 ## 3.2 Scene Models and Views
 
 ### 3.2.1. Model files
  
-1. Enable "artist control" e.g. material specs, transparency/blending... ?
-2. OBJ fixes/improvements (#896 - multiple items!)
-3. Assimp support ? [2.1]
-4. Improved AWD support/bug fixes ? [2.1]  (#1185, #1465, #1468)
-5. MD5 child naming (#1454) [2.1]
+1. ~Enable "artist control" e.g. material specs, transparency/blending... ?~ Proper loader support (#2) will take care of this.
+2. Assimp support
```

Comment:
Does this mean CH will no longer support OBJ, AWD, or MD5? I.e. should we close issues #896, #1185, #1465, #1468, and #1454 if Assimp is the new go-to model file format?

Reply:
Assimp is a library which supports something like 40 different formats, OBJ and MD5 among them. I have not completely written off AWD either, but it has become a lower priority. I want to keep the tickets open to make sure the new implementation is free of those bugs.

Reply:
Ah, not just another model file format. Makes sense now.  So we would basically throw out most of the current Raj parsers and use assimp to populate object data/scenes?

Reply:
Correct. My original plan was to interface Assimp through an API in a similar spirit to the current Loader API. Custom formats or other formats not supported in Assimp could be implemented through this API (like they are now). AWD would be a good example. Of course, adding them to Assimp is always a possibility too, but that means C++ code so personnel to do that is limited.

Reply:
Ah, assimp is native, so I'm guessing we'll need a JNI/JNA wrapper between the new loader API you're suggesting and assimp proper?

Reply:
Correct. A JNI wrapper is also required to access many vendor specific extensions so I have already begun the work that will make this possible.

Reply:
Have we considered moving object data/scenes loading into a separate module?
I would prefer to keep native dependencies out of the core.

Reply:
That was already planned, though not spelled out. I will update it. The core will only contain the java interface that all loaders will access/implement. This ultimately applies to the extensions too.